### PR TITLE
LZ-LB-item-details

### LINKED ISSUE #13: 13. As a user, I want to view details of my purchases to better understand my purchase patterns
AC:

* From the shopping list, user can tap to navigate to a detailed view of the item
* Detail view includes the following information:
  * Name of the item
  * Last purchased date
  * Estimated number of days until the next purchase
  * Number of times the item has been purchased"
* When in the “item details” view, a back arrow should appear in the header that takes the user back to the “list” view

### PR DESCRIPTION
**Pull request by lilyhoratio**
## Description

This PR reuses the modal component to display item details when the item is clicked.

## Related Issue

Closes #13 

## Acceptance Criteria

AC:

From the shopping list, user can tap to navigate to a detailed view of the item
Detail view includes the following information:
Name of the item
Last purchased date
Estimated number of days until the next purchase
Number of times the item has been purchased"
When in the “item details” view, a back arrow should appear in the header that takes the user back to the “list” view --> Decided to use modal, not needed

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### After
Screen Shot 2020-07-14 at 5 23 18 PM

## Testing Steps / QA Criteria

1. On the list view, click on the name of the item which should pop up a modal containing 3 bullet points
2. The modal should disappear when the "x" is clicked

### CODE REVIEW

File: src/components/Item.js [hunk 1]
```diff
@@ -42,21 +57,41 @@ const Item = ({ item, handleChange, deleteItem }) => {
...
+        </button>
+      </li>
+      {showItemInfo && (
+        <Modal setDisplay={setShowItemInfo}>
+          <h2>{item.name}</h2>
+          <ul>
+            <li>Last purchased on {lastPurchasedDate}</li>
+            <li>Next purchase in {item.nextPurchase} days</li>
+            <li>
+              {item.numberOfPurchases > 1
```
Comment (segdeha) [inline src/components/Item.js:87]: 👏👏👏 I went looking for whether you handled both singular and plural cases and was impressed that you thought of it!

File: src/styles/Item.css [hunk 2]
```diff
@@ -19,6 +19,7 @@
   border-radius: 5px;
   margin-bottom: 10px;
   height: 35px;
+  cursor: pointer;
```
Comment (segdeha) [inline src/styles/Item.css:22]: Nice UX touch!

File: src/components/Item.js [hunk 3]
```diff
@@ -42,21 +57,41 @@ const Item = ({ item, handleChange, deleteItem }) => {
...
+        <span className={className} onClick={handleModalChange}>
+          {item.name}
+        </span>
+
+        <button className="delete" onClick={() => deleteItem(item.id)}>
+          Delete
+        </button>
+      </li>
+      {showItemInfo && (
+        <Modal setDisplay={setShowItemInfo}>
```
Comment (segdeha) [inline src/components/Item.js:81]: While the feature appears to work fine in the browser, I believe having `<Modal/>` alongside the `<li/>` results in invalid HTML when the modal is open. The modal is a `<div/>` and the only element that is valid to be at the same level as an `<li/>` is another `<li/>`. The fact that it works is the browser being generous to us! 😅 

What I would suggest (though maybe as a refactor later) would be moving `<Modal/>` outside/after the `<ul/>`, though I'm not sure if that will make it more tricky to track which item you're modaling 🤔

Comment (segdeha) [reply src/components/Item.js:81]: Hmm, 🤔 about it more, another thing you could try that would be simpler than factoring that out to after the list would be to put it just inside the `</li>`, though I'm not 💯% sure that will work

Comment (Lukabaramishvili) [reply src/components/Item.js]: Thank you for pointing this out. This comment totally re-created my respect for `</ul>` and `</li>` elements 😁. 

Moving `Modal` inside the `<li>` element worked perfectly. I tried refactoring too but putting it inside `<li>` saves extra iteration in the List component. Thanks for the advice 🙂.

**Pull request merged**